astropy/io/fits/tests/test_hdulist.py

```diff
@@ -913,3 +914,38 @@ def test_proper_error_raised_on_non_fits_file_with_unicode(self):
         # This should raise an IOError because there is no end card.
         with pytest.raises(IOError):
             fits.open(filename)
+
+    # Avoid a spurious pyflakes warning about an unused import
+    # of six by adding a reason string
+    @pytest.mark.skipif(six.PY2, reason='six.PY2')
+    @pytest.mark.parametrize('ignore', [False, True])
+    @pytest.mark.parametrize('fileobj', [False, True])
+    def test_no_resource_warning_raised_on_non_fits_file(self, ignore, fileobj):
+        """
+        Regression test for https://github.com/astropy/astropy/issues/6168
+
+        The ResourceWarning shows up when (in python 3+) you try to
+        open a non-FITS file when using a filename.
+        """
+
+        filename = self.temp('not-fits.fits')
```

Comment:
The original issue mentioned `*.dat` file but here you are testing with `*.fits`. Probably does not matter (if FITS code does not impose extension hard-coding) but I would like to know if this test actually fails in current `master` *without* this PR.